### ISSUE TITLE
libsqlite3-sys always recompiles on Windows when bundled feature is used

### ISSUE THREAD

**Issue by LunarLambda** [rusqlite/rusqlite#1331]
Cargo.toml:
```toml
[dependencies]
rusqlite = { version = "0.29.0", features = ["bundled"] }
```

`cargo build` takes about a minute flat for minute changes because it's recompiling/relinking libsqlite3-sys. (work machine with a very aggressive AV...)

A similar issue was fixed in https://github.com/rusqlite/rusqlite/pull/627 but doesn't seem to apply to the `bundled` feature.
I believe this can be fixed using the appropriate `cargo:rerun-if-changed` directives, but I only briefly glanced at the build script.

**Comment by gwenn** [comment]
Cannot reproduce:
```sh
$ cargo check
   Compiling libsqlite3-sys v0.26.0
    Checking rusqlite v0.29.0
    Checking rusqlite-example v0.1.0 (...\rusqlite-example)
    Finished dev [unoptimized + debuginfo] target(s) in 25.33s
$ cargo check
    Finished dev [unoptimized + debuginfo] target(s) in 0.06s
```

**Comment by LunarLambda** [comment]
what about `cargo build`? I didn't try with cargo check.

**Comment by gwenn** [comment]
Cannot reproduce (Rust 1.68 (GNU 64-bit)):
```sh
$ cargo build
...
   Compiling libsqlite3-sys v0.26.0
...
   Compiling rusqlite v0.29.0
   Compiling rusqlite-example v0.1.0 (C:\Users\treguiergw\perso\rusqlite-example)
    Finished dev [unoptimized + debuginfo] target(s) in 9.85s
$ cargo build
    Finished dev [unoptimized + debuginfo] target(s) in 0.07s
$ cargo -V
cargo 1.68.0 (115f34552 2023-02-26)
```

**Comment by LunarLambda** [comment]
strange. maybe the AV is removing files it shouldn't. I'll close until I can find a reproduction.

**Comment by samgqroberts** [comment]
I'm late here, but I was still running into these same symptoms on MacOS despite #627 / #435 all being resolved and I found a solution that might help someone else. The core issue I found did not have anything to do with libsqlite3-sys / rusqlite (I also encountered this issue with objc-sys / tauri), but was a general bad interaction between eager rust-analyzer runs and cargo commands. The issue was the combination of the following:

1. I run rust-analyzer on every file save (via VSCode)
2. rust-analyzer in some way disagrees with cargo on how exactly to compile sys crates, like libsqlite3-sys here, or at least on how to detect whether they are out of date in the build output cache
3. rust-analyzer and the cargo commands I was using shared a build output directory (`target/debug/`)

So what would happen is I would run `cargo test` (or whatever cargo command), cargo would compile libsqlite3-sys the way it liked into `target/debug/`, I would edit a file in VSCode, rust-analyzer would run on file save, rust-analyzer would determine the libsqlite3-sys cached in `target/debug/` was out of date somehow, it would recompile that and everything else on top of it, I'd go back to running `cargo test`, it would think libsqlite3-sys was out of date, so on so forth.

The solution I found was to make sure rust-analyzer was not using the same build output directory as my cargo commands, and for the VSCode-embedded rust-analyzer that meant adding `"rust-analyzer.cargo.targetDir": true` to `.vscode/settings.json`, which makes rust-analyzer use its own `target/rust-analyzer/` directory. It uses more disk space, but the compilation time improvements are so worth it for me. Bonus: cargo test and rust-analyzer generally don't get in each other's way anymore, so I'm not stuck looking at `Blocking waiting for file lock on package cache` nearly as much anymore.
